src/main/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4Kit.kt

```diff
@@ -31,6 +31,9 @@ class GoogleAnalyticsFirebaseGA4Kit : KitIntegration(), KitIntegration.EventList
         context: Context
     ): List<ReportingMessage>? {
         Logger.info("$name Kit relies on a functioning instance of Firebase Analytics. If your Firebase Analytics instance is not configured properly, this Kit will not work")
+        currentUser?.consentState?.let {
+            setConsent(currentUser.consentState)
+        }
```

Comment:
if you wanted to be extremely cautious - you should grab a reference and check if the reference is null. otherwise in your code it's possible that the `currentuser` or the `consentstate` becomes null (due to another thread modifying it) in between lines. not sure if this compile but here's the gist of how i would do it:
```suggestion
        val user = currentUser?.consentState?
        user?.let {
            setConsent(user)
        }
```